integration/setup/teardown.go

```diff
@@ -19,7 +20,9 @@ func Teardown(c Config) error {
 	// We don't wait for the resources to be deleted, since we are going to delete them below.
 	err = c.Host.G8sClient().ProviderV1alpha1().AzureConfigs(c.Host.TargetNamespace()).Delete(clusterID, &metav1.DeleteOptions{})
 	if err != nil {
-		return microerror.Mask(err)
+		if !errors.IsNotFound(err) {
```

Comment:
This was needed because in the new test we delete the `CustomResource` before hand. I don't think this change affects anything else, and allow us to avoid erroring when the `CustomResource` is not there.